### PR TITLE
Remove IllegalStateException in BuckManifestFactory

### PR DESCRIPTION
**Pull request by kageiit**
### Overview

Addresses https://github.com/robolectric/robolectric/commit/afb9f22ca8eb8f098db4a69c045f10d04b4d02b6#commitcomment-25143180

### Proposed Changes

Fixes the logic in BuckManifestFactory to correctly handle assets and resources even if they are not of the same size

### CODE REVIEW

**Comment by kageiit** [comment]
cc @xian @jrodbx @cwoodwar6

File: robolectric/src/main/java/org/robolectric/internal/BuckManifestFactory.java [hunk 1]
```diff
@@ -33,28 +33,21 @@ public ManifestIdentifier identify(Config config) {
...
     } else {
-      resDir = buckResources.get(pathCount - 1);
-      assetsDir = buckAssets.get(pathCount - 1);
       libraries = new ArrayList<>();
 
-      for (int i = 0; i < pathCount - 1; i++) {
-        libraries.add(new ManifestIdentifier(null, null,
-            buckResources.get(i), buckAssets.get(i), null));
+      for (FsFile buckResource: buckResources) {
+        libraries.add(new ManifestIdentifier(null, null, buckResource, null, null));
```
Comment (jrodbx) [inline robolectric/src/main/java/org/robolectric/internal/BuckManifestFactory.java]: CI is failing due to ambiguous constructor reference.  See here:
https://github.com/robolectric/robolectric/blob/master/robolectric/src/main/java/org/robolectric/internal/ManifestIdentifier.java#L17,L32

Comment (xian) [inline robolectric/src/main/java/org/robolectric/internal/BuckManifestFactory.java]: Yep, needs `(String) null`. Otherwise LGTM.

**Comment by xian** [comment]
```java
org.robolectric.internal.BuckManifestFactoryTest > multiple_res_dirs FAILED
    org.junit.ComparisonFailure: expected:<[2]> but was:<[5]>
        at org.robolectric.internal.BuckManifestFactoryTest.multiple_res_dirs(BuckManifestFactoryTest.java:65)

org.robolectric.internal.BuckManifestFactoryTest > pass_multiple_res_dirs_in_file FAILED
    org.junit.ComparisonFailure: expected:<[2]> but was:<[5]>
        at org.robolectric.internal.BuckManifestFactoryTest.pass_multiple_res_dirs_in_file(BuckManifestFactoryTest.java:91)
```

**Comment by kageiit** [comment]
Lemme fix the tests while at it :)

**Comment by xian** [comment]
Also merged to `robolectric-3.5.x` branch.

**Pull request merged**